models/vision_language_model.py

```diff
@@ -97,12 +101,4 @@ def generate(self, input_ids, image, attention_mask=None, max_new_tokens=5):
     def load_checkpoint(self, path):
         print(f"Loading weights from full VLM checkpoint: {path}")
         checkpoint = torch.load(path, map_location=torch.device('cuda' if torch.cuda.is_available() else 'cpu'))        
-        self.load_state_dict(checkpoint)
-
-    @classmethod
-    def from_pretrained(cls, cfg):
-        model = cls(cfg)
-        model.vision_encoder = ViT.from_pretrained(cfg)
-        model.decoder = LanguageModel.from_pretrained(cfg)
-
-        return model
\ No newline at end of file
+        self.load_state_dict(checkpoint)
```

Comment:
With this PR we would be able to get the `from_pretrained` API, do you think we would need this anymore?

I understand that keeping this would be a good backup to have, but thinking out aloud here.